tests/integration/telemetry/util/util.go

```diff
@@ -99,6 +103,43 @@ func PromDump(cluster cluster.Cluster, prometheus prometheus.Instance, query pro
 	return ""
 }
 
+// QueryPrometheus queries prometheus and returns the result once the query stabilizes
+func QueryPrometheus(t framework.TestContext, cluster cluster.Cluster, query prometheus.Query, promInst prometheus.Instance) (string, error) {
+	t.Helper()
+	t.Logf("query common with: %v", query)
+
+	val, err := promInst.Query(cluster, query)
+	if err != nil {
+		return "", err
+	}
+	got, err := prometheus.Sum(val)
+	if err != nil {
+		t.Logf("value: %s", val.String())
+		return "", fmt.Errorf("could not find metric value: %v", err)
+	}
+	t.Logf("get value %v", got)
+	return val.String(), nil
+}
+
+func ValidateMetric(t framework.TestContext, cluster cluster.Cluster, prometheus prometheus.Instance, query prometheus.Query, want float64) {
```

Comment:
with this no longer in `prometheus`, I suggest we update the name to be `ValidatePrometheusMetric`. Thoughts?